File: L02_3/app/src/main/java/com/example/l02_3/MainActivity.java
package com.example.l02_3;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.net.Uri;
import android.os.Bundle;
import android.view.View;
import android.widget.EditText;
import android.widget.TextView;
import android.widget.Toast;

public class MainActivity extends AppCompatActivity {

    int request_Code = 1;
    boolean start = true;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
    }
    public void onClick(View view) {
        Intent i = new Intent(this, SecondActivity.class);
        Bundle bundle = new Bundle();
        EditText actText = (EditText)findViewById(R.id.txtUsername);
        bundle.putString("key", actText.getText().toString());
        i.putExtras(bundle);
        startActivityForResult(i, request_Code);
    }

    public void onActivityResult(int requestCode, int resultCode, Intent data) {
        super.onActivityResult(requestCode, resultCode, data);
        if (requestCode == request_Code) {
            if (resultCode == RESULT_OK) {
                String res = data.getData().toString();
                EditText actText = (EditText)findViewById(R.id.txtUsername);
                actText.setText(res);
                Toast.makeText(this, res, Toast.LENGTH_SHORT).show();
            }
        }
    }

}

File: L04_2/app/src/main/java/com/example/l04_2/MainActivity.kt
package com.example.l04_2

import android.app.Activity
import android.content.Intent
import android.os.Bundle
import android.view.View
import android.widget.EditText
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity


class MainActivity : AppCompatActivity() {

    private val requestCode = 10;
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
    }

    fun onClick(view: View) {
        val intent = Intent(this, SecondActivity::class.java)
        val bundle = Bundle();
        val textBox: EditText = findViewById(R.id.txtBox)
        bundle.putString("kotlin_key", textBox.text.toString())
        intent.putExtras(bundle)
        startActivityForResult(intent, requestCode)
    }

    override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?) {
        super.onActivityResult(requestCode, resultCode, data)
        if (requestCode == requestCode) {
            if (resultCode == Activity.RESULT_OK) {
                val res = data!!.data.toString()
                val newTextBox: EditText = findViewById(R.id.txtBox)
                newTextBox.setText(res);
                Toast.makeText(this, res, Toast.LENGTH_SHORT).show()
            }
        }
    }

}
